addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -735,6 +737,15 @@ function makeActionManager(env) {
                     dialogCloseResolve();
                 }
             }
+            get componentProps() {
+                const componentProps = { ...this.props };
+                if (this.constructor.Component === View) {
+                    componentProps.__beforeLeave__ = this.__beforeLeave__;
+                    componentProps.__getGlobalState__ = this.__getGlobalState__;
+                    componentProps.__getLocalState__ = this.__getLocalState__;
```

Comment:
what about having a single props `__callbackRecorders__` that is an object with all these? That way, no need to have the callbackRecorderNames constant in View

Reply:
we will create another system maybe. Callback recorders are not really appreciated...